src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,500 @@
+            var annotatedExpressionWithinDocument = (TExpressionSyntax)annotatedSemanticDocument.Root.GetAnnotatedNodesAndTokens(annotatedExpression).Single().AsNode()!;
+            // var methodSymbolInfo = GetMethodSymbolFromExpression(annotatedSemanticDocument, annotatedExpressionWithinDocument, cancellationToken);
+
+            var updatedSolutionWithParameter = await AddParameterToMethodHeaderAsync(annotatedSemanticDocument, annotatedExpressionWithinDocument, parameterName, cancellationToken).ConfigureAwait(false);
+            var updatedSemanticDocument = await SemanticDocument.CreateAsync(updatedSolutionWithParameter.GetDocument(document.Document.Id), cancellationToken).ConfigureAwait(false);
+
+            var newExpression = (TExpressionSyntax)updatedSemanticDocument.Root.GetAnnotatedNodesAndTokens(annotatedExpression).Single().AsNode()!;
+            var documentWithUpdatedMethodBody = ConvertExpressionWithNewParameter(updatedSemanticDocument, newExpression, parameterName, allOccurrences, cancellationToken);
+            return documentWithUpdatedMethodBody.Project.Solution;
+        }
+
+        protected static Task<Solution> AddParameterToMethodHeaderAsync(SemanticDocument document, TExpressionSyntax expression, string parameterName, CancellationToken cancellationToken)
+        {
+            var invocationDocument = document.Document;
+            var syntaxFacts = invocationDocument.GetRequiredLanguageService<ISyntaxFactsService>();
+            var methodExpression = expression.FirstAncestorOrSelf<SyntaxNode>(node => syntaxFacts.isMethodDeclaration(node), ascendOutOfTrivia: true)!;
+            var semanticModel = document.SemanticModel;
+            var symbolInfo = (IMethodSymbol)semanticModel.GetRequiredDeclaredSymbol(methodExpression, cancellationToken);
+            var parameterType = semanticModel.GetTypeInfo(expression, cancellationToken).Type ?? document.SemanticModel.Compilation.ObjectType;
+            var refKind = syntaxFacts.GetRefKindOfArgument(expression);
+
+            return AddParameterService.Instance.AddParameterAsync(
+                invocationDocument,
+                symbolInfo,
+                parameterType,
+                refKind,
+                parameterName,
+                newParameterIndex: null,
+                fixAllReferences: false,
+                cancellationToken);
+        }
+
+        protected static string GetNewParameterName(SemanticDocument document, TExpressionSyntax expression, CancellationToken cancellationToken)
+        {
+            var invocationDocument = document.Document;
+            var semanticModel = document.SemanticModel;
+            var semanticFacts = invocationDocument.GetRequiredLanguageService<ISemanticFactsService>();
+            return semanticFacts.GenerateNameForExpression(
+                    semanticModel, expression, capitalize: false, cancellationToken: cancellationToken);
```

Comment:
indentation wrong.

Reply:
do you need the named prameter for cancellationToken?

Reply:
will fix and no